src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -75,12 +84,38 @@ public long amount() {
         throw new UnsupportedOperationException("amount() not yet implemented");
     }
 
-    // @todo #15:30min Implement prefix() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
-    public String prefix() {
-        throw new UnsupportedOperationException("prefix() not yet implemented");
+    public String prefix() throws Exception {
+        try {
+            final String prefix = new ItemAt<>(
+                new SplitText(
+                    new TextOf(
+                        this.transaction
+                    ),
+                    new TextOf(";")
+                ).iterator(),
+                //@checkstyle MagicNumberCheck (1 line)
+                3
+            ).value().asString();
+            //@checkstyle MagicNumberCheck (1 line)
+            if (prefix.length() < 8 || prefix.length() > 32) {
+                throw new IllegalArgumentException("Invalid prefix size");
+            }
+            final Pattern pattern = Pattern.compile(
+                //@checkstyle LineLengthCheck (1 line)
+                "^([A-Za-z0-9+\\/]{4})*([A-Za-z0-9+\\/]{4}|[A-Za-z0-9+\\/]{3}=|[A-Za-z0-9+\\/]{2}==)$"
+            );
+            if (!pattern.matcher(prefix).matches()) {
+                throw new IllegalArgumentException("Invalid base64 prefix");
+            }
+            return prefix;
+            //@checkstyle IllegalCatchCheck (1 line)
+        } catch (final Exception exception) {
+            throw new IllegalArgumentException(
```

Comment:
@paulodamaso this catch block is problematic. We are catching all exceptions and wrapping it saying that the prefix string is wrong. That may not be the case. We are already checking the validity of it above, so what scenario here would be the result of a badly formatted string? I notice the bounds check is missing, better handle it by checking the resulting list has the right element than this sloppy approach.

Reply:
@carlosmiranda Added test for a bad formatted transaction string.